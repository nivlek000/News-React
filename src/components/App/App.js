import React, { Component } from 'react';
import logo from '../../logo.svg';
import './App.css';
//redux
import { connect } from 'react-redux';
//redux actions
import * as newsActions from '../../store/actions/news.action.js';

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>

        {this.props.bla.data.map(item => (
          <div key={item.id}>
            <h3>{item.title}</h3>
            <button onClick={() => { console.log(item.title + ' is being deleted'); this.delNews(item) }}>Delete</button>
          </div>
        )
        )}

        <input ref={(i => this.inputRef = i)} />
        <button onClick={() => { console.log(this.inputRef.value); this.addNews() }}>Add Headline</button>

        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }

  //redux actions
  addNews = () => {
    return this.props.dispatch(newsActions.addNews({
      id: this.props.bla.data.length + 1,
      title: this.inputRef.value
    }))
  }

  delNews = (news) => {
    return this.props.dispatch(newsActions.delNews(news))
  }



}







//redux
function mapStateToProps(state) {
  return {
    bla: state.news
  };
}

export default connect(mapStateToProps)(App); //connect fron redux is a higher order function
