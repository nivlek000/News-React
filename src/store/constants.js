export const ADD_NEWS = 'ADD_NEWS'; 
export const DEL_NEWS = 'DEL_NEWS';