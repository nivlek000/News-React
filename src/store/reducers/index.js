import { combineReducers } from 'redux'; //destructuring
import news from './news';
import users from './users';


export default combineReducers({
    news: news,
    users

})