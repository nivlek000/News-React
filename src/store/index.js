import { createStore } from "redux";
import reducers from './reducers';

const storee=createStore(
    reducers,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

//export default createStore(reducers)
export default storee