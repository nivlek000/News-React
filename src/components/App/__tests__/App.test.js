import React from 'react';
import { mount } from 'enzyme';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import store from '../../../store';

import App from '../App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <App />
    </Provider>, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App ', () => {
  describe('Snapshot', () => {
    it('should render app component with children', () => {
      const component = renderer.create(
        <Provider store={store}>
          <App />
        </Provider>
      ).toJSON()
      expect(component).toMatchSnapshot();
    });
  });

  describe('Unit', () => {
    it('should display 2 news', () => {
      const component = mount(
        <Provider store={store}>
          <App />
        </Provider>
      );

      expect(component.find('.news')).toHaveLength(0);
    });
  });
});

