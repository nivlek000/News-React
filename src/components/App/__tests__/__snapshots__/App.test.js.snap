// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App  Snapshot should render app component with children 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <div>
    <h3>
      3million richer snake
    </h3>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "border": "1px solid",
          "color": "red",
          "cursor": "pointer",
          "margin": "3px",
          "padding": "2px 10px",
        }
      }
    >
      Delete
    </button>
  </div>
  <div>
    <h3>
      Monkey swallows snake
    </h3>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "border": "1px solid",
          "color": "red",
          "cursor": "pointer",
          "margin": "3px",
          "padding": "2px 10px",
        }
      }
    >
      Delete
    </button>
  </div>
  <hr />
  <input />
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "aliceblue",
        "border": "1px solid",
        "color": "blue",
        "cursor": "pointer",
        "margin": "3px",
        "padding": "2px 10px",
      }
    }
  >
    Add Headline
  </button>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;
